libcodechecker/server/session_manager.py

```diff
@@ -472,10 +406,57 @@ def get_max_run_count(self):
         """
         return self.__max_run_count
 
+    def _get_local_session_from_db(self, token):
+        """
+        Creates a local session if a valid session token can be found in the
+        database.
+        """
+
+        if not self.__database_connection:
+            return
+
+        try:
+            transaction = self.__database_connection()
+            db_record = transaction.query(SessionRecord) \
+                .filter(SessionRecord.token == token) \
+                .limit(1).one_or_none()
+
+            if db_record:
+                if (datetime.now() - db_record.last_access). \
+                        total_seconds() <= \
+                        self.__auth_config['session_lifetime']:
+                    # If a token was found in the database and the session
+                    # for it can still be resurrected, we create a local
+                    # session.
+                    user_name = db_record.user_name
+                    system_permission = transaction.query(SystemPermission) \
+                        .filter(SystemPermission.name == user_name) \
+                        .limit(1).one_or_none()
+
+                    is_root = True if system_permission else False
+                    local_session = _Session(
+                        token, user_name,
+                        db_record.groups.split(';'),
+                        self.__auth_config['session_lifetime'],
+                        is_root, self.__database_connection)
+
+                    return local_session
+                else:
+                    # The token has expired, remove it from the database.
```

Comment:
I would put token validity check and removal into a separate method. The method name ` _get_local_session_from_db` does not reflect that.